src/persist-client/src/impl/machine.rs

```diff
@@ -758,15 +761,86 @@ mod tests {
 
             let state = Arc::new(Mutex::new(DatadrivenState::default()));
             let cpu_heavy_runtime = Arc::new(CpuHeavyRuntime::new());
+            let write = Arc::new(Mutex::new(
```

Comment:
do we need this here? I intentionally made the existing ones short lived. if we do need it, let's push it into DatadrivenState

Reply:
with some of the other changes... maybe not! I can't remember the exact motivation for it, so as the other pieces move around I'll doublecheck